src/ophys_etl/modules/segmentation/merge/roi_merging.py

```diff
@@ -0,0 +1,378 @@
+    timeseries_lookup = {}
+
+    # ROIs that are too large to be considered valid
+    anomalous_rois = {}
+
+    logger.info(f'initially {len(roi_lookup)} ROIs')
+    while keep_going:
+
+        # will be set to True if a merger occurs
+        keep_going = False
+
+        # statistics on this pass for INFO messages
+        n0 = len(roi_lookup)
+
+        # fill in sub-video lookup for ROIs that changed
+        # during the last iteration
+        for roi_id in roi_lookup:
+            if roi_id in sub_video_lookup:
+                continue
+            roi = roi_lookup[roi_id]
+            sub_video_lookup[roi_id] = sub_video_from_roi(roi,
+                                                          video_data)
+
+        # find all pairs of ROIs that abut
+        raw_merger_candidates = set()
+        for roi_id_0 in neighbor_lookup:
+            for roi_id_1 in neighbor_lookup[roi_id_0]:
+                if roi_id_0 > roi_id_1:
+                    pair = (roi_id_0, roi_id_1)
+                else:
+                    pair = (roi_id_1, roi_id_0)
+                raw_merger_candidates.add(pair)
+
+        # only need to calculate metrics for those that have
+        # changed since the last iteration, though
+        raw_merger_candidates = list(raw_merger_candidates)
+        merger_candidates = []
+        for pair in raw_merger_candidates:
+            if pair not in merger_to_metric:
+                merger_candidates.append(pair)
+
+        shuffler.shuffle(merger_candidates)
+
+        local_t0 = time.time()
+        timeseries_lookup = update_timeseries_lookup(
+                              merger_candidates,
+                              timeseries_lookup,
+                              sub_video_lookup,
+                              filter_fraction=filter_fraction,
+                              n_processors=n_processors)
+
+        logger.info('updated timeseries lookup '
+                    f'in {time.time()-local_t0:.2f} seconds')
+
+        local_t0 = time.time()
+        self_corr_lookup = create_self_corr_lookup(
+                               merger_candidates,
+                               sub_video_lookup,
+                               timeseries_lookup,
+                               filter_fraction,
+                               n_processors)
+
+        new_merger_metrics = get_merger_metric_from_pairs(
+                                   merger_candidates,
+                                   sub_video_lookup,
+                                   timeseries_lookup,
+                                   self_corr_lookup,
+                                   filter_fraction,
+                                   n_processors)
+
+        logger.info('updated self_corr and calculated metrics in '
+                    f'{time.time()-local_t0:.2f} seconds')
+
+        for pair in new_merger_metrics:
+            metric = new_merger_metrics[pair]
+            if metric >= -1.0*corr_acceptance:
```

Comment:
Because the merger metric is a median z-score from a sample, the -1 allows the child ROI to be lower than the mean of the parent ROI's distribution. If corr_acceptance=1, we are saying "merge the child ROI if it's median correlation is 1-sigma lower than the parent ROI's mean correlation or higher". If the median z-score is positive, it means that the child ROI is somehow more correlated with the parent ROI than the parent ROI's own pixels are with itself (which, in this case, is also taken as an indication for merger).